app/code/Magento/ConfigurableProduct/Model/Product/Type/Configurable.php

```diff
@@ -1284,9 +1285,11 @@ public function isPossibleBuyFromList($product)
      * Returns array of sub-products for specified configurable product
      * Result array contains all children for specified configurable product
      *
-     * @param \Magento\Catalog\Model\Product $product
+     * @param Product $product
      * @param array $requiredAttributeIds Attributes to include in the select; one-dimensional array
      * @return ProductInterface[]
+     * @deprecated since method doesn't consider skip stock filter param
+     * @see getUsedProductsStockFilter
      */
     public function getUsedProducts($product, $requiredAttributeIds = null)
     {
```

Comment:
Hello @engcom-Charlie,
thank you for your contribution!

I have a question! According to https://devdocs.magento.com/contributor-guide/backward-compatible-development/ you must deprecate the method and add a new method with the new parameter(s) instead of adding them to a public method but in this specific case (since the parameter is an optional one and it's not backward-incompatible) (Indeed following at 100% contributor guidelines this PR should be submitted on Magento 2.5 release line) wouldn't it be a better option, change:

public function getUsedProducts($product, $requiredAttributeIds = null)

in

getUsedProducts($product, $requiredAttributeIds = null, ?bool $skipStock = false)

and

$collection = $this->getConfiguredUsedProductCollection($product, false, $requiredAttributeIds);

in

$collection = $this->getConfiguredUsedProductCollection($product, $skipStock, $requiredAttributeIds);

so:

we shouldn't change all occurrences of getUsedProducts to getUsedProductsStockFilter and all third-party vendors shouldn't change occurrence

A QA/Team Leader ( @sivaschenko ) could help us with this choice?